dhall-nix/release.nix

```diff
@@ -210,6 +121,9 @@ in
         testProject = dhallToNix ''
           λ(r : { foo : Bool, bar : Text, baz : Natural }) → r.{ foo, bar }
         '';
+        testShadow = dhallToNix ''
+          λ(x : Natural) → λ(x1 : Natural) → λ(x : Natural) → x@1
```

Comment:
Could you add another testcase like 

```
λ(x : Natural) → λ(x1 : Natural) → λ(x : Natural) → [ x@1, x1 ]
```

to see whether the existing reference to `x1` is properly preserved?